### PR TITLE
pre-commit test

### PR DESCRIPTION
**Pull request by PyRo1121**
This is a start place for pre-commit being implemented and shows a few errors to correct and a reason I love this tool so much.

`pylint...................................................................Failed
- hook id: pylint
- exit code: 30

************* Module setup
setup.py:1:0: C0114: Missing module docstring (missing-module-docstring)
setup.py:20:21: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
************* Module src.revChatGPT
src/revChatGPT/__init__.py:1:0: C0103: Module name "revChatGPT" doesn't conform to snake_case naming style (invalid-name)
************* Module src.revChatGPT.__main__
src/revChatGPT/__main__.py:30:0: C0301: Line too long (89/80) (line-too-long)
src/revChatGPT/__main__.py:85:0: C0301: Line too long (82/80) (line-too-long)
src/revChatGPT/__main__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
src/revChatGPT/__main__.py:8:0: E0401: Unable to import 'revChatGPT.revChatGPT' (import-error)
src/revChatGPT/__main__.py:11:0: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/__main__.py:11:14: W0621: Redefining name 'prompt' from outer scope (line 46) (redefined-outer-name)
src/revChatGPT/__main__.py:43:8: R1722: Consider using sys.exit() (consider-using-sys-exit)
src/revChatGPT/__main__.py:46:8: C0103: Constant name "prompt" doesn't conform to UPPER_CASE naming style (invalid-name)
src/revChatGPT/__main__.py:48:12: R1724: Unnecessary "elif" after "continue", remove the leading "el" from "elif" (no-else-continue)
src/revChatGPT/__main__.py:80:12: C0103: Constant name "lines_printed" doesn't conform to UPPER_CASE naming style (invalid-name)
src/revChatGPT/__main__.py:98:19: W0703: Catching too general exception Exception (broad-except)
src/revChatGPT/__main__.py:107:19: W0703: Catching too general exception Exception (broad-except)
************* Module src.revChatGPT.revChatGPT
src/revChatGPT/revChatGPT.py:32:0: C0301: Line too long (85/80) (line-too-long)
src/revChatGPT/revChatGPT.py:51:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:130:0: C0301: Line too long (100/80) (line-too-long)
src/revChatGPT/revChatGPT.py:134:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:217:0: C0301: Line too long (87/80) (line-too-long)
src/revChatGPT/revChatGPT.py:221:0: C0301: Line too long (113/80) (line-too-long)
src/revChatGPT/revChatGPT.py:337:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:338:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:353:0: C0301: Line too long (107/80) (line-too-long)
src/revChatGPT/revChatGPT.py:361:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:368:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:376:0: C0301: Line too long (93/80) (line-too-long)
src/revChatGPT/revChatGPT.py:386:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:394:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:410:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:412:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:433:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:435:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:459:0: C0301: Line too long (112/80) (line-too-long)
src/revChatGPT/revChatGPT.py:460:0: C0301: Line too long (84/80) (line-too-long)
src/revChatGPT/revChatGPT.py:465:0: C0301: Line too long (107/80) (line-too-long)
src/revChatGPT/revChatGPT.py:473:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:474:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:476:0: C0301: Line too long (84/80) (line-too-long)
src/revChatGPT/revChatGPT.py:496:0: C0301: Line too long (110/80) (line-too-long)
src/revChatGPT/revChatGPT.py:501:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:502:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:504:0: C0301: Line too long (82/80) (line-too-long)
src/revChatGPT/revChatGPT.py:521:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:523:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:526:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:567:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:1:0: C0114: Missing module docstring (missing-module-docstring)
src/revChatGPT/revChatGPT.py:1:0: C0103: Module name "revChatGPT" doesn't conform to snake_case naming style (invalid-name)
src/revChatGPT/revChatGPT.py:11:0: E0401: Unable to import 'requests' (import-error)
src/revChatGPT/revChatGPT.py:12:0: E0401: Unable to import 'tls_client' (import-error)
src/revChatGPT/revChatGPT.py:15:0: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:20:0: C0115: Missing class docstring (missing-class-docstring)
src/revChatGPT/revChatGPT.py:36:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:41:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:62:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:88:12: W0702: No exception type(s) specified (bare-except)
src/revChatGPT/revChatGPT.py:81:16: W0702: No exception type(s) specified (bare-except)
src/revChatGPT/revChatGPT.py:92:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:94:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
src/revChatGPT/revChatGPT.py:122:15: W0703: Catching too general exception Exception (broad-except)
src/revChatGPT/revChatGPT.py:145:12: W0702: No exception type(s) specified (bare-except)
src/revChatGPT/revChatGPT.py:166:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:182:8: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
src/revChatGPT/revChatGPT.py:189:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:193:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:194:8: R1720: Unnecessary "elif" after "raise", remove the leading "el" from "elif" (no-else-raise)
src/revChatGPT/revChatGPT.py:206:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
src/revChatGPT/revChatGPT.py:238:19: W0703: Catching too general exception Exception (broad-except)
src/revChatGPT/revChatGPT.py:193:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
src/revChatGPT/revChatGPT.py:248:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:289:0: C0115: Missing class docstring (missing-class-docstring)
src/revChatGPT/revChatGPT.py:517:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:560:4: C0116: Missing function or method docstring (missing-function-docstring)

------------------------------------------------------------------
Your code has been rated at 7.59/10 (previous run: 7.59/10, +0.00)`

### CODE REVIEW

**Comment by PyRo1121** [comment]
Added pylint too. This can be configured further adding args of your choice

File: src/revChatGPT/__main__.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
-from revChatGPT.revChatGPT import Chatbot
+from __future__ import annotations
```
Comment (acheong08) [inline src/revChatGPT/__main__.py:1]: Is this a dependency I need to add to the requirements? I don't know much about annotations sorry

Comment (PyRo1121) [reply src/revChatGPT/__main__.py:1]: Let me test everything works. These were changes made purely from running test and I didn't personally touch the code. I will keep you updated on the running portion.

File: src/revChatGPT/__main__.py [hunk 2]
```diff
@@ -22,15 +25,17 @@ def get_input(prompt):
...
     Repo: github.com/acheong08/ChatGPT
-    """)
+    """,
+    )
     print("Type '!help' to show commands")
     print("Press enter twice to submit your question.\n")
 
     if exists("config.json"):
-        with open("config.json", "r", encoding='utf-8') as f:
+        with open("config.json", encoding="utf-8") as f:
```
Comment (acheong08) [inline src/revChatGPT/__main__.py:38]: Is it not better to keep the 'r' to make sure it's only reading from it?

**Review by acheong08 — COMMENTED** [review]
The rest of the formatting changes look good to me

**Comment by acheong08** [comment]
> src/revChatGPT/revChatGPT.py:36:4: C0116: Missing function or method docstring (missing-function-docstring)

Documentation is most certainly lacking here. I will probably write another repo to go through each function and send it off to ChatGPT to add comments haha

File: src/revChatGPT/__main__.py (hunk 2 above)
Comment (PyRo1121) [reply src/revChatGPT/__main__.py:38]: I am going to fine tune this I apologize. You keep studying for exams and I will post a comment once ready. I hate to distract you from that. I will be working on this for a bit this evening.

File: src/revChatGPT/__main__.py (hunk 1 above)
Comment (PyRo1121) [reply src/revChatGPT/__main__.py:1]: To clarify the change here it looks like it is following [PEP-563](https://peps.python.org/pep-0563/)

File: src/revChatGPT/__main__.py (hunk 2 above)
Comment (acheong08) [reply src/revChatGPT/__main__.py:38]: > You keep studying for exams

I just completed the last exam for this week a few minutes ago. I'm free to work on this now (though there are still internal assessments coming up. I will probably do those with ChatGPT as an assistant)

File: src/revChatGPT/__main__.py (hunk 1 above)
Comment (acheong08) [reply src/revChatGPT/__main__.py:1]: > To clarify the change here it looks like it is following [PEP-563](https://peps.python.org/pep-0563/)

Seems to be a built in library. I won't worry about it then

Comment (PyRo1121) [reply src/revChatGPT/__main__.py:1]: I see the issue all though it is accurate it is not needed I had a arg from another project of mine in the config still.

File: src/revChatGPT/__main__.py (hunk 2 above)
Comment (PyRo1121) [reply src/revChatGPT/__main__.py:38]: I have narrowed this down to be a accurate fix as it was formatted under autopep8. I am testing now to see what rule of pep8 it is formatting to correct and comply with.

File: src/revChatGPT/__main__.py (hunk 1 above)
Comment (acheong08) [reply src/revChatGPT/__main__.py:1]: If it's unnecessary please remove it. Minimal dependencies are recommended for the main code.

**Comment by PyRo1121** [comment]
Most recent run for reference 
`pre-commit run --all-files                  
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...............................................................Passed
debug statements (python)................................................Passed
fix double quoted strings................................................Failed
- hook id: double-quote-string-fixer
- exit code: 1
- files were modified by this hook

Fixing strings in setup.py
Fixing strings in src/revChatGPT/__main__.py
Fixing strings in src/revChatGPT/revChatGPT.py

python tests naming..................................(no files to check)Skipped
fix requirements.txt.................................(no files to check)Skipped
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted setup.py
reformatted src/revChatGPT/__main__.py
reformatted src/revChatGPT/revChatGPT.py

All done! ✨ 🍰 ✨
3 files reformatted, 2 files left unchanged.

Reorder python imports...................................................Passed
Add trailing commas......................................................Passed
pyupgrade................................................................Passed
autopep8.................................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

src/revChatGPT/__main__.py:28:80: E501 line too long (89 > 79 characters)
src/revChatGPT/__main__.py:83:80: E501 line too long (82 > 79 characters)
src/revChatGPT/revChatGPT.py:32:80: E501 line too long (85 > 79 characters)
src/revChatGPT/revChatGPT.py:51:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:81:17: E722 do not use bare 'except'
src/revChatGPT/revChatGPT.py:88:13: E722 do not use bare 'except'
src/revChatGPT/revChatGPT.py:130:80: E501 line too long (100 > 79 characters)
src/revChatGPT/revChatGPT.py:134:80: E501 line too long (86 > 79 characters)
src/revChatGPT/revChatGPT.py:145:13: E722 do not use bare 'except'
src/revChatGPT/revChatGPT.py:217:80: E501 line too long (87 > 79 characters)
src/revChatGPT/revChatGPT.py:221:80: E501 line too long (113 > 79 characters)
src/revChatGPT/revChatGPT.py:337:80: E501 line too long (88 > 79 characters)
src/revChatGPT/revChatGPT.py:338:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:353:80: E501 line too long (107 > 79 characters)
src/revChatGPT/revChatGPT.py:361:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:368:80: E501 line too long (86 > 79 characters)
src/revChatGPT/revChatGPT.py:376:80: E501 line too long (93 > 79 characters)
src/revChatGPT/revChatGPT.py:386:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:394:80: E501 line too long (86 > 79 characters)
src/revChatGPT/revChatGPT.py:410:80: E501 line too long (88 > 79 characters)
src/revChatGPT/revChatGPT.py:412:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:433:80: E501 line too long (88 > 79 characters)
src/revChatGPT/revChatGPT.py:435:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:459:80: E501 line too long (112 > 79 characters)
src/revChatGPT/revChatGPT.py:460:80: E501 line too long (84 > 79 characters)
src/revChatGPT/revChatGPT.py:465:80: E501 line too long (107 > 79 characters)
src/revChatGPT/revChatGPT.py:466:80: E501 line too long (80 > 79 characters)
src/revChatGPT/revChatGPT.py:473:80: E501 line too long (88 > 79 characters)
src/revChatGPT/revChatGPT.py:474:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:476:80: E501 line too long (84 > 79 characters)
src/revChatGPT/revChatGPT.py:496:80: E501 line too long (110 > 79 characters)
src/revChatGPT/revChatGPT.py:501:80: E501 line too long (88 > 79 characters)
src/revChatGPT/revChatGPT.py:502:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:504:80: E501 line too long (82 > 79 characters)
src/revChatGPT/revChatGPT.py:521:80: E501 line too long (88 > 79 characters)
src/revChatGPT/revChatGPT.py:523:80: E501 line too long (117 > 79 characters)
src/revChatGPT/revChatGPT.py:526:80: E501 line too long (86 > 79 characters)
src/revChatGPT/revChatGPT.py:567:80: E501 line too long (117 > 79 characters)

mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

src/revChatGPT/__main__.py:77: error: Need type annotation for "messages" (hint: "messages: List[<type>] = ...")  [var-annotated]
src/revChatGPT/__main__.py:82: error: Need type annotation for "formatted_parts" (hint: "formatted_parts: List[<type>] = ...")  [var-annotated]
src/revChatGPT/revChatGPT.py:21: error: Module "json" is not valid as a type  [valid-type]
src/revChatGPT/revChatGPT.py:21: note: Perhaps you meant to use a protocol matching the module structure?
src/revChatGPT/revChatGPT.py:37: error: Incompatible types in assignment (expression has type "None", variable has type "str")  [assignment]
src/revChatGPT/revChatGPT.py:42: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:43: error: Unsupported target for indexed assignment (json?)  [index]
src/revChatGPT/revChatGPT.py:44: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:45: error: Unsupported target for indexed assignment (json?)  [index]
src/revChatGPT/revChatGPT.py:49: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:62: error: The return type of a generator function should be "Generator" or one of its supertypes  [misc]
src/revChatGPT/revChatGPT.py:96: error: Incompatible types in assignment (expression has type "Dict[Any, Any]", variable has type "CaseInsensitiveDict[Union[str, bytes]]")  [assignment]
src/revChatGPT/revChatGPT.py:98: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:101: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:108: error: json? has no attribute "get"  [attr-defined]
src/revChatGPT/revChatGPT.py:110: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:111: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:120: error: Incompatible types in assignment (expression has type "str", variable has type "Response")  [assignment]
src/revChatGPT/revChatGPT.py:121: error: Value of type "Response" is not indexable  [index]
src/revChatGPT/revChatGPT.py:125: error: Item "None" of "Optional[Match[str]]" has no attribute "group"  [union-attr]
src/revChatGPT/revChatGPT.py:148: error: Invalid index type "str" for "str"; expected type "Union[SupportsIndex, slice]"  [index]
src/revChatGPT/revChatGPT.py:155: error: Argument 1 to "loads" has incompatible type "Response"; expected "Union[str, bytes]"  [arg-type]
src/revChatGPT/revChatGPT.py:156: error: Value of type "Response" is not indexable  [index]
src/revChatGPT/revChatGPT.py:157: error: Value of type "Response" is not indexable  [index]
src/revChatGPT/revChatGPT.py:158: error: Value of type "Response" is not indexable  [index]
src/revChatGPT/revChatGPT.py:166: error: Invalid type comment or annotation  [valid-type]
src/revChatGPT/revChatGPT.py:193: error: Missing return statement  [return]
src/revChatGPT/revChatGPT.py:195: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:196: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:197: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:200: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:202: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:203: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:207: error: json? has no attribute "get"  [attr-defined]
src/revChatGPT/revChatGPT.py:209: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:210: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:215: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:226: error: Unsupported target for indexed assignment (json?)  [index]
src/revChatGPT/revChatGPT.py:229: error: Unsupported target for indexed assignment (json?)  [index]
src/revChatGPT/revChatGPT.py:235: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:237: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:242: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:244: error: Return value expected  [return-value]
src/revChatGPT/revChatGPT.py:252: error: json? has no attribute "__iter__" (not iterable)  [attr-defined]
src/revChatGPT/revChatGPT.py:253: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:255: error: Value of type json? is not indexable  [index]
src/revChatGPT/revChatGPT.py:256: error: Argument 4 to "OpenAIAuth" has incompatible type "Optional[Any]"; expected "str"  [arg-type]
src/revChatGPT/revChatGPT.py:266: error: Unsupported target for indexed assignment (json?)  [index]
src/revChatGPT/revChatGPT.py:275: error: Unsupported target for indexed assignment (json?)  [index]
src/revChatGPT/revChatGPT.py:278: error: Unsupported target for indexed assignment (json?)  [index]
src/revChatGPT/revChatGPT.py:295: error: Incompatible default for argument "proxy" (default has type "None", argument has type "str")  [assignment]
src/revChatGPT/revChatGPT.py:295: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
src/revChatGPT/revChatGPT.py:295: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
src/revChatGPT/revChatGPT.py:305: error: Incompatible types in assignment (expression has type "None", variable has type "str")  [assignment]
src/revChatGPT/revChatGPT.py:449: error: Invalid type comment or annotation  [valid-type]
src/revChatGPT/revChatGPT.py:538: error: "List[Any]" has no attribute "split"  [attr-defined]
src/revChatGPT/revChatGPT.py:540: error: Argument "access_token" to "save_access_token" of "OpenAIAuth" has incompatible type "List[Any]"; expected "str"  [arg-type]
Found 54 errors in 2 files (checked 5 source files)

pylint...................................................................Failed
- hook id: pylint
- exit code: 30

************* Module setup
setup.py:1:0: C0114: Missing module docstring (missing-module-docstring)
setup.py:20:21: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
************* Module src.revChatGPT
src/revChatGPT/__init__.py:1:0: C0103: Module name "revChatGPT" doesn't conform to snake_case naming style (invalid-name)
************* Module src.revChatGPT.__main__
src/revChatGPT/__main__.py:28:0: C0301: Line too long (89/80) (line-too-long)
src/revChatGPT/__main__.py:83:0: C0301: Line too long (82/80) (line-too-long)
src/revChatGPT/__main__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
src/revChatGPT/__main__.py:6:0: E0401: Unable to import 'revChatGPT.revChatGPT' (import-error)
src/revChatGPT/__main__.py:9:0: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/__main__.py:9:14: W0621: Redefining name 'prompt' from outer scope (line 44) (redefined-outer-name)
src/revChatGPT/__main__.py:41:8: R1722: Consider using sys.exit() (consider-using-sys-exit)
src/revChatGPT/__main__.py:44:8: C0103: Constant name "prompt" doesn't conform to UPPER_CASE naming style (invalid-name)
src/revChatGPT/__main__.py:46:12: R1724: Unnecessary "elif" after "continue", remove the leading "el" from "elif" (no-else-continue)
src/revChatGPT/__main__.py:78:12: C0103: Constant name "lines_printed" doesn't conform to UPPER_CASE naming style (invalid-name)
src/revChatGPT/__main__.py:96:19: W0703: Catching too general exception Exception (broad-except)
src/revChatGPT/__main__.py:105:19: W0703: Catching too general exception Exception (broad-except)
************* Module src.revChatGPT.revChatGPT
src/revChatGPT/revChatGPT.py:32:0: C0301: Line too long (85/80) (line-too-long)
src/revChatGPT/revChatGPT.py:51:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:130:0: C0301: Line too long (100/80) (line-too-long)
src/revChatGPT/revChatGPT.py:134:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:217:0: C0301: Line too long (87/80) (line-too-long)
src/revChatGPT/revChatGPT.py:221:0: C0301: Line too long (113/80) (line-too-long)
src/revChatGPT/revChatGPT.py:337:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:338:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:353:0: C0301: Line too long (107/80) (line-too-long)
src/revChatGPT/revChatGPT.py:361:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:368:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:376:0: C0301: Line too long (93/80) (line-too-long)
src/revChatGPT/revChatGPT.py:386:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:394:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:410:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:412:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:433:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:435:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:459:0: C0301: Line too long (112/80) (line-too-long)
src/revChatGPT/revChatGPT.py:460:0: C0301: Line too long (84/80) (line-too-long)
src/revChatGPT/revChatGPT.py:465:0: C0301: Line too long (107/80) (line-too-long)
src/revChatGPT/revChatGPT.py:473:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:474:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:476:0: C0301: Line too long (84/80) (line-too-long)
src/revChatGPT/revChatGPT.py:496:0: C0301: Line too long (110/80) (line-too-long)
src/revChatGPT/revChatGPT.py:501:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:502:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:504:0: C0301: Line too long (82/80) (line-too-long)
src/revChatGPT/revChatGPT.py:521:0: C0301: Line too long (88/80) (line-too-long)
src/revChatGPT/revChatGPT.py:523:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:526:0: C0301: Line too long (86/80) (line-too-long)
src/revChatGPT/revChatGPT.py:567:0: C0301: Line too long (117/80) (line-too-long)
src/revChatGPT/revChatGPT.py:1:0: C0114: Missing module docstring (missing-module-docstring)
src/revChatGPT/revChatGPT.py:1:0: C0103: Module name "revChatGPT" doesn't conform to snake_case naming style (invalid-name)
src/revChatGPT/revChatGPT.py:11:0: E0401: Unable to import 'requests' (import-error)
src/revChatGPT/revChatGPT.py:12:0: E0401: Unable to import 'tls_client' (import-error)
src/revChatGPT/revChatGPT.py:15:0: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:20:0: C0115: Missing class docstring (missing-class-docstring)
src/revChatGPT/revChatGPT.py:36:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:41:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:62:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:88:12: W0702: No exception type(s) specified (bare-except)
src/revChatGPT/revChatGPT.py:81:16: W0702: No exception type(s) specified (bare-except)
src/revChatGPT/revChatGPT.py:92:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:94:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
src/revChatGPT/revChatGPT.py:122:15: W0703: Catching too general exception Exception (broad-except)
src/revChatGPT/revChatGPT.py:145:12: W0702: No exception type(s) specified (bare-except)
src/revChatGPT/revChatGPT.py:166:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:182:8: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
src/revChatGPT/revChatGPT.py:189:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:193:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:194:8: R1720: Unnecessary "elif" after "raise", remove the leading "el" from "elif" (no-else-raise)
src/revChatGPT/revChatGPT.py:206:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
src/revChatGPT/revChatGPT.py:238:19: W0703: Catching too general exception Exception (broad-except)
src/revChatGPT/revChatGPT.py:193:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
src/revChatGPT/revChatGPT.py:248:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:289:0: C0115: Missing class docstring (missing-class-docstring)
src/revChatGPT/revChatGPT.py:517:4: C0116: Missing function or method docstring (missing-function-docstring)
src/revChatGPT/revChatGPT.py:560:4: C0116: Missing function or method docstring (missing-function-docstring)

------------------------------------------------------------------
Your code has been rated at 7.58/10 (previous run: 7.59/10, -0.01)`

File: setup.py [hunk 3]
```diff
@@ -1,4 +1,8 @@
-from setuptools import setup, find_packages
+from __future__ import annotations
```
Comment (acheong08) [inline setup.py:1]: So this is necessary correct?

Comment (PyRo1121) [reply setup.py:1]: :| I missed that one my apologies when I refactored the yaml and corrected __main__.py

**Review by acheong08 — APPROVED** [review]
Looks good. Will merge and test

File: setup.py (hunk 3 above)
Comment (PyRo1121) [reply setup.py]: Corrected and also removed from 'revChatGPT.py' as I missed on that one as well

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (acheong08) [inline .pre-commit-config.yaml:1]: Should this be in .github?

Comment (acheong08) [reply .pre-commit-config.yaml:1]: nvm sorry

**Pull request merged**